src/reanimated2/shareables.ts

```diff
@@ -273,9 +273,16 @@ export function makeShareableCloneOnUIRecursive<T>(
       typeof value === 'function'
     ) {
       if (isHostObject(value)) {
-        return value as FlatShareableRef<T>;
+        // We call `_makeShareableClone` because provided HostObject
+        // might not be a Shareable.
+        return _makeShareableClone(
+          _makeShareableClone(value)
```

Comment:
Why do we call `_makeShareableClone` twice?

Reply:
Yes, I accidentally comitted it 😅